### ISSUE TITLE
BatchNorm layer appears broken in multi-backend Keras

### ISSUE THREAD

**Issue by martin-gorner** [keras-team/keras-core#338]
classic MNIST training:
- [Keras multi-backend with BN on JAX](https://colab.research.google.com/drive/1ZZI6UK9umgCDSIRcPqVUl6YpLmRdwMF3?resourcekey=0-8YS9usjjxUuh0FUl_JW0Ow&usp=sharing). Final accuracy __19%__!!!
- [Keras multi-backend w/o BN on JAX](https://colab.research.google.com/drive/16Yj63zZHmH2oQNBtNYmP5W0KJJ_RnLQi?resourcekey=0-Ut-aMYTyo3i36iJeQB4LRQ&usp=sharing). Final accuracy: 99%
- [Keras multi-backend with BN on TF](https://colab.research.google.com/drive/1ipSnQuGe2w4KGphG6hMeWdzt3H8m36cx?resourcekey=0-ZH-RRNE3rfXqqurX8yE_YA&usp=sharing). Final accuracy: __14%__!!!
- [Keras multi-backend w/o BN on TF](https://colab.research.google.com/drive/1beSnSJDvBwqYXzaGpvdVH9bUgCM2Jtsi?resourcekey=0-jmZOshPvFOUqWOEGq-z2Dg&usp=sharing). Final accuracy: 99%
- [Keras baseline (regular tf.keras on TF, with BN)](https://colab.research.google.com/drive/1dI3RTfVvYdYER09Q6onYIdsh3vkRmEci?resourcekey=0-FgN08JmB4pWDpsL8aePs8A&usp=sharing). Final accuracy: 99%

The common factors triggering the issue seem to be: using the BatchNorm layer, on Keras multi-backend. The choice of backend does not seem to change the behavior.
